backend/services/esweService.ts

```diff
@@ -98,42 +102,61 @@ export default class EsweService {
 
   async getLearnerLatestAssessments(nomisId: string): Promise<LearnerLatestAssessments> {
     if (!app.esweEnabled) {
-      return {
-        enabled: app.esweEnabled,
-        content: {},
-      }
+      return createFlaggedContent({})
     }
 
-    const content = await this.curiousApi.getLearnerLatestAssessments(nomisId)
+    const context = await this.systemOauthClient.getClientCredentialsTokens()
+    const learnerLatestAssessments = await this.curiousApi.getLearnerLatestAssessments(context, nomisId)
 
-    const filterSkillsAndGetLatestGrade = (functionalSkillLevels, skillToFilter) =>
-      functionalSkillLevels[0].qualifications
-        .filter((functionalSkillLevel) => functionalSkillLevel.qualification.qualificationType === skillToFilter)
-        .sort(
-          (a, b) =>
-            new Date(b.qualification.assessmentDate).getTime() - new Date(a.qualification.assessmentDate).getTime()
-        )[0]
+    const compareByDateDesc = (a: curious.LearnerAssessment, b: curious.LearnerAssessment) =>
```

Comment:
Not sure you meant to make a duplicate of this function?

Reply:
Good catch, thanks 👍

Reply:
Ahh it turns out this was intentional.

They have the same name but exist in two separate methods. They sort by different properties on the LearnerEducation type and pass to the reused method `compareByDate`.

The first one in `getLatestLearningDifficulty` sorts by the learning start date to determine the latest learning difficulty
```
const compareByDateDesc = (a: curious.LearnerEducation, b: curious.LearnerEducation) =>
        compareByDate(parseDate(a.learningStartDate), parseDate(b.learningStartDate))
```

The second one in `getLearnerLatestAssessments` sorts by `qualification.assessmentDate` to get the latest assessment for a course

```
    const compareByDateDesc = (a: curious.LearnerAssessment, b: curious.LearnerAssessment) =>
      compareByDate(parseDate(a.qualification.assessmentDate), parseDate(b.qualification.assessmentDate))
```